spec/requests/place_spec.rb

```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Places API', type: :request do
+  # initialize test data
+  let!(:places) { create_list(:place, 10) }
+  let(:first_place) { places.first }
+  let(:place_id) { places.first.id }
+  let(:place_name) { places.first.name.downcase }
+
+  # Test suite for GET /places
+  describe 'GET /places' do
+    # make HTTP get request before each example
+    before { get '/places' }
+
+    it 'returns places' do
+      # Note `json` is a custom helper to parse JSON responses
+      expect(json['places']).not_to be_empty
+      expect(json['places'].size).to eq(10)
+    end
+
+    it 'returns status code 200' do
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  # # Test suite for GET /places?q=<place name>
+  # describe 'GET /places?q=<place name>' do
+
+  #   before { get "/places?q=#{place_name}" }
+
+  #   context 'when the record exists' do
+  #     it 'returns a list with the place' do
+  #       expect(json['places']).not_to be_empty
+  #       expect(json['places'][0]['name']).to eq(place_name)
+  #     end
+
+  #     it 'returns status code 200' do
+  #       expect(response).to have_http_status(200)
+  #     end
+  #   end
+
+  #   context 'when the record does not exist' do
+  #     let(:place_name) { 'not-here' }
+
+  #     it 'returns empty list' do
+  #       expect(json['places']).to be_empty
+  #     end
+
+  #     it 'returns status code 200' do
+  #       expect(response).to have_http_status(200)
+  #     end
+  #   end
+  # end
+
+  # Test suite for GET /places/:id
+  describe 'GET /places/:id' do
+    before do
+      # Create a tag for this place
+      first_place.tags << Tag.create!(name: 'tag1')
+      # Create a review for this place
+      Review.create!(username: 'me', description: 'nice', 'score': 4, place: first_place)
+      Review.create!(username: 'you', description: 'not nice', 'score': 2, place: first_place)
+
+      get "/places/#{place_id}"
+    end
+
+    context 'when the record exists' do
+      it 'returns the place' do
+        expect(json).not_to be_empty
+        expect(json['place']['id']).to eq(place_id)
+        # Check tags
+        expect(json['tags'].length).to eq(1)
+        expect(json['tags'][0]).to eq('tag1')
+        # Check reviews
+        expect(json['latest_reviews'].length).to eq(2)
+      end
+
+      it 'returns status code 200' do
+        expect(response).to have_http_status(200)
```

Comment:
instead of the status code - you can use the symbols, :ok, :created, :not_found. . . 
http://www.railsstatuscodes.com
[official docs](https://guides.rubyonrails.org/layouts_and_rendering.html#options-for-render)